src/db/collections/outputs/indexer/mod.rs

```diff
@@ -0,0 +1,218 @@
+                    None,
+                )
+                .await?
+                .map_ok(|doc| bson::from_document::<OutputResult>(doc).unwrap())
+                .try_collect::<Vec<_>>()
+                .await?;
+            Ok(Some(OutputsResult { ledger_index, outputs }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Creates indexer output indexes.
+    pub async fn create_indexer_output_indexes(&self) -> Result<(), Error> {
+        let collection = self.0.collection::<OutputDocument>(OutputDocument::COLLECTION);
+
+        collection
+            .create_index(
+                IndexModel::builder()
+                    .keys(doc! { "output.kind": 1 })
+                    .options(IndexOptions::builder().name("output_kind_index".to_string()).build())
+                    .build(),
+                None,
+            )
+            .await?;
+
+        collection
+            .create_index(
+                IndexModel::builder()
+                    .keys(doc! { "output.alias_id": 1 })
+                    .options(
+                        IndexOptions::builder()
+                            .name("output_alias_id_index".to_string())
+                            .partial_filter_expression(
+                                doc! { "output.alias_id": { "$exists": true }, "metadata.spent": null },
+                            )
+                            .build(),
+                    )
+                    .build(),
+                None,
+            )
+            .await?;
+
+        collection
+            .create_index(
+                IndexModel::builder()
+                    .keys(doc! { "output.foundry_id": 1 })
+                    .options(
+                        IndexOptions::builder()
+                            .name("output_foundry_id_index".to_string())
+                            .partial_filter_expression(
+                                doc! { "output.foundry_id": { "$exists": true }, "metadata.spent": null },
+                            )
+                            .build(),
+                    )
+                    .build(),
+                None,
+            )
+            .await?;
+
+        collection
+            .create_index(
+                IndexModel::builder()
+                    .keys(doc! { "output.nft_id": 1 })
+                    .options(
+                        IndexOptions::builder()
+                            .name("output_nft_id_index".to_string())
+                            .partial_filter_expression(
+                                doc! { "output.nft_id": { "$exists": true }, "metadata.spent": null },
+                            )
+                            .build(),
+                    )
+                    .build(),
+                None,
+            )
+            .await?;
+
+        collection
+            .create_index(
+                IndexModel::builder()
+                    .keys(doc! { "output.unlock_conditions": 1 })
```

Comment:
I don't think we have that field anymore.

Reply:
oop

Reply:
Wait, no, this isn't ledger updates.